iceoryx2-bb/log/src/lib.rs

```diff
@@ -187,6 +208,7 @@ pub fn set_log_level(v: LogLevel) {
 
 /// Returns the current log level
 pub fn get_log_level() -> u8 {
+    LOG_LEVEL.store(*ENV_LOG_LEVEL as u8, Ordering::Relaxed);
```

Comment:
Doesn't this prevent to set the log level programmatically?

I think you need to wrap this in a similar fashion like `INIT.call_once` and also do the call in `set_log_level`, similar to how it is done in `set_logger` but inside and outside of the `call_once` call.

Reply:
@elBoberido yes, calling it in `get_log_level` will override any value being set programmatically.
But do you meant calling it inside `get_logger` for once? I tried your proposal inside `get_log_level` but the init was not successful, but when i had it in `get_logger` just after `LOGGER` init, it seems to be the right place for init.

```
pub fn get_logger() -> &'static dyn Logger {
    INIT.call_once(|| {
        unsafe { LOGGER = Some(&*DEFAULT_LOGGER) };
        println!("setting env var");
        LOG_LEVEL.store(*ENV_LOG_LEVEL as u8, Ordering::Relaxed);
    });

    unsafe { *LOGGER.as_ref().unwrap() }
}

```

Reply:
Oh, I was not clear enough. You need to create a second static variable, e.g. `INIT_LOG_LEVEL`. The `INIT` variable could also be renamed to `INIT_LOGGER`.

So, no matter if `set_log_level` or `get_log_level` is called first, the `LOG_LEVEL` variable will be initialized only once with the entry from the env variable.

Reply:
@elBoberido Can you take a look on the last commit? I hope this is what you were asking for :)

Reply:
@elBoberido I moved the call to the functions mentioned above, now `INIT_LOG_LEVEL` is done in `set_log_level` and `get_log_level`

Reply:
@elBoberido I tested the changes on my terminal, even with setting an unknown trace level like `blabla` and printed a warning message to the user, and set the loglevel to default `Info` 
where should I document the new feature for debug LogLevel? FAQ file has a paragraph on setting log level programatically, should this get into the README.MD file?

Reply:
@zmostafa looks good. I think the FAQ would be a good candidate for this.

Could you please:
- also look into the other open discussion
- rebase to that we can trigger the CI
- try what happens when `IOX2_LOG_LEVEL` is set in the `env` section of `.cargo/config.toml` and also set as env variable in the terminal?
  - if the env variable set via the terminal has a higher priority than when set via `.cargo/config.toml`, then we could set the log level in `.cargo/config.toml` to trace, so that all iceoryx developer get the detailed log by default but can still control it by setting a custom level in `.bashrc`

Reply:
@elBoberido Branch is rebased.
setting IOX2_LOG_LEVEL in terminal takes precedence over setting it in `.cargo/config.toml` I added the env variable to `.cargon/config.toml` and configured it to Trace, also mentioned this precedence in FAQ.md

Reply:
@zmostafa it seems there is still a merge conflict in `iceoryx2-bb/log/src/lib.rs`. Can you do another rebase?

Reply:
@zmostafa okay, the branch is no mergeable. The markdown linter is now happy, though